File: JavaScript/Prefix Sum/1732. Find the Highest Altitude.js
/* 1732. Find the Highest Altitude
Difficulty: Easy
Submission Link: https://leetcode.com/problems/find-the-highest-altitude/submissions/1379773462/ */

var largestAltitude = function (gain) {
    let sum = 0;
    let alt = [];
    let max;
    for (let i = 0; i <= gain.length; i++) {
        alt.push(sum);
        sum = sum + gain[i];
    }
    max = alt[0];
    for (let j = 0; j < alt.length; j++) {
        if (max < alt[j]) {
            max = alt[j];
        }
    }
    return max;
}

File: Python3/Prefix Sum/1732. Find the Highest Altitude.py
""" 1732. Find the Highest Altitude
Difficulty: Easy
Submission Link: https://leetcode.com/problems/find-the-highest-altitude/submissions/1379773561/ """

class Solution:
    def largestAltitude(self, gain: list[int]) -> int:
        sum = 0
        max_alt = 0
        
        for g in gain:
            sum += g
            max_alt = max(max_alt, sum)
        
        return max_alt